test/e2e-go/restAPI/simulate/simulateRestAPI_test.go

```diff
@@ -218,6 +218,136 @@ func TestSimulateTransaction(t *testing.T) {
+	a.NoError(err)
+	clearState := ops.Program
+
+	txn, err := testClient.MakeUnsignedApplicationCallTx(
+		0, nil, nil, nil,
+		nil, nil, transactions.NoOpOC,
+		approval, clearState, basics.StateSchema{}, basics.StateSchema{}, 0,
+	)
+	a.NoError(err)
+	txn, err = testClient.FillUnsignedTxTemplate(senderAddress, 1, 1001, 0, txn)
+	a.NoError(err)
+	stxn, err := testClient.SignTransactionWithWallet(wh, nil, txn)
+	a.NoError(err)
+
+	// Get controller for follower node
+	followControl, err := fixture.GetNodeController("Follower")
+	a.NoError(err)
+	followClient := fixture.GetAlgodClientForController(followControl)
+
+	// Set sync round on follower
+	followerSyncRound := uint64(4)
+	err = followClient.SetSyncRound(followerSyncRound)
+	a.NoError(err)
+
+	cfg, err := config.LoadConfigFromDisk(followControl.GetDataDir())
+	a.NoError(err)
+
+	// Let the primary node make some progress
+	primaryClient := fixture.GetAlgodClientForController(nc)
+	err = fixture.ClientWaitForRoundWithTimeout(primaryClient, followerSyncRound+uint64(cfg.MaxAcctLookback))
+	a.NoError(err)
+
+	// Let follower node progress as far as it can
+	err = fixture.ClientWaitForRoundWithTimeout(followClient, followerSyncRound+uint64(cfg.MaxAcctLookback)-1)
+	a.NoError(err)
+
+	simulateRequest := v2.PreEncodedSimulateRequest{
+		TxnGroups: []v2.PreEncodedSimulateRequestTransactionGroup{
+			{
+				Txns: []transactions.SignedTxn{stxn},
+			},
+		},
+	}
+
+	// Simulate transactions against the follower node
+	simulateTransactions := func(request v2.PreEncodedSimulateRequest) (result v2.PreEncodedSimulateResponse, err error) {
+		encodedRequest := protocol.EncodeReflect(&request)
+		var resp []byte
+		resp, err = followClient.RawSimulateRawTransaction(encodedRequest)
+		if err != nil {
+			return
+		}
+		err = protocol.DecodeReflect(resp, &result)
+		return
+	}
+
+	// Test default behavior (should use latest round available)
+	result, err := simulateTransactions(simulateRequest)
+	a.NoError(err)
+	a.Len(result.TxnGroups, 1)
+	a.Empty(result.TxnGroups[0].FailureMessage)
+	a.Len(result.TxnGroups[0].Txns, 1)
+	a.NotNil(result.TxnGroups[0].Txns[0].Txn.Logs)
+	a.Len(*result.TxnGroups[0].Txns[0].Txn.Logs, 1)
+	a.Equal(followerSyncRound+uint64(cfg.MaxAcctLookback), binary.BigEndian.Uint64((*result.TxnGroups[0].Txns[0].Txn.Logs)[0]))
+
+	// Test with previous rounds
+	for i := uint64(0); i < cfg.MaxAcctLookback; i++ {
+		simulateRequest.Round = basics.Round(followerSyncRound + i)
+		result, err = simulateTransactions(simulateRequest)
+		a.NoError(err)
+		a.Len(result.TxnGroups, 1)
+		a.Empty(result.TxnGroups[0].FailureMessage)
+		a.Len(result.TxnGroups[0].Txns, 1)
+		a.NotNil(result.TxnGroups[0].Txns[0].Txn.Logs)
+		a.Len(*result.TxnGroups[0].Txns[0].Txn.Logs, 1)
+		a.LessOrEqual(followerSyncRound+i+1, binary.BigEndian.Uint64((*result.TxnGroups[0].Txns[0].Txn.Logs)[0]))
+	}
+
+	// There should be a failure when the round is too far back
+	simulateRequest.Round = basics.Round(followerSyncRound - 3)
```

Comment:
@Eric-Warehime I found out something unexpected about follower node behavior when I wrote this test.

I would have expected the first round at which ledger information would be unavailable would be `followerSyncRound-1`, but I was only able to trigger a failure by asking for `followerSyncRound-3`.

Is this to be expected? (I'll add that when I manually spun up a follower node and performed this test, it failed on `followerSyncRound-2`, so maybe there are some timing or test environment abnormalities happening in this test).

Reply:
The error you're looking for, `round %d before dbRound` is from https://github.com/algorand/go-algorand/blob/master/ledger/acctupdates.go#L1401 which is checking if the requested round is below the first db round. And `cachedDBRound` would, after enough rounds go by, stop at `syncRound-1`. So it would make sense to only see this error on `syncRound-2`.